src/hooks/useCrop.ts

```diff
@@ -0,0 +1,377 @@
+import { useReducer, useCallback } from 'react';
+import { clamp } from '../helpers/index';
+
+// https://developer.mozilla.org/en-US/docs/Web/CSS/cursor //
+export type ICursorStyles = (
```

Comment:
Well in the case of type vs interfaces, both look pretty similar but they have different capabilities.
It's not that one is new and the other is old rather than depends in the intention.

Typescript will throw an error if a type is decelerated and the same name type already exist but allows to have multiple declarations of interfaces with the same name and merges them.

https://thisthat.dev/interface-vs-type/

Although even the page recommends the usage of interface over types, there are scenarios where type is necessary.  In this case specific case, since I'm declaring specific "string" values I need it to use a type to declare them. This case is similar to design and sizes in buttons.

I haven't seen it used much in the base code, but I guess I could try enums for this case.
https://blog.logrocket.com/writing-readable-code-with-typescript-enums-a84864f340e9/

Just as a comment aside,  another occasion to use types is when you need to join more than one interface, because is not posible to extend more than one, but you can join more than one.
https://blog.logrocket.com/types-vs-interfaces-in-typescript/

By no means I'm an expert on the subject and  I've seen on internet that there's not a great agreement about it's usage, just hundreds of examples of how people use them.